File: ConsoleApplication1/AppUser.cpp
#include "stdafx.h"
#include "AppUser.h"

#include "DeilTask.h"
#include <iostream>

AppUser::AppUser()
{
}


AppUser::~AppUser()
{
}

void AppUser::StartWork()
{
	DeilTask deilTask(this);
	deilTask.Run();
}

void AppUser::DoProgress(float value)
{
	std::cout << value << std::endl;
}

File: ConsoleApplication2/AppUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication2
{
    class AppUser : IProcess
    {
        public void StartWork()
        {
            DeilTask task = new DeilTask(this);
            task.Run();
        }
        public void DoProcess(float value)
        {
            Console.WriteLine(value);
        }
    }
}
